torch/nn/modules/module.py

```diff
@@ -755,27 +786,70 @@ def named_parameters(self, memo=None, prefix=''):
-        for mname, module in self.named_children():
-            submodule_prefix = prefix + ('.' if prefix else '') + mname
-            for name, p in module.named_parameters(memo, submodule_prefix):
-                yield name, p
+        gen = self._named_members(
+            lambda module: module._parameters.items(),
+            memo=memo, prefix=prefix, recurse=recurse)
+        for elem in gen:
+            yield elem
+
+    def buffers(self, memo=None, recurse=True):
+        r"""Returns an iterator over module buffers.
+
+        Args:
+            memo (set): if specified, excludes the enclosed buffers from
+                being yielded. This check is performed using ``id()``.
+                This method adds the yielded buffers to ``memo``.
+            recurse (bool): if True, then yields buffers of this module
+                and all submodules. Otherwise, yields only buffers that
+                are direct members of this module.
+
+        Yields:
+            torch.Tensor: module buffer
+
+        Example::
+
+            >>> for buf in model.buffers():
+            >>>     print(type(buf.data), buf.size())
+            <class 'torch.FloatTensor'> (20L,)
+            <class 'torch.FloatTensor'> (20L, 1L, 5L, 5L)
+
+        """
+        for name, buf in self.named_buffers(memo=memo, recurse=recurse):
+            yield buf
+
+    def named_buffers(self, memo=None, prefix='', recurse=True):
+        r"""Returns an iterator over module buffers, yielding both the
+        name of the buffer as well as the buffer itself
+
+        Args:
+            memo (set): if specified, excludes the enclosed buffers from
+                being yielded. This check is performed using ``id()``.
+                This method adds the yielded buffers to ``memo``.
+            prefix (str): prefix to prepend to all buffer names.
+            recurse (bool): if True, then yields buffers of this module
+                and all submodules. Otherwise, yields only buffers that
+                are direct members of this module.
+
+        Yields:
+            (string, torch.Tensor): Tuple containing the name and buffer
+
+        Example::
+
+            >>> for name, buf in self.named_buffers():
+            >>>    if name in ['running_var']:
+            >>>        print(buf.size())
+
+        """
+        gen = self._named_members(
+            lambda module: module._buffers.items(),
+            memo=memo, prefix=prefix, recurse=recurse)
+        for elem in gen:
+            yield elem
 
     def _all_buffers(self, memo=None):
-        if memo is None:
-            memo = set()
-        for name, b in self._buffers.items():
-            if b is not None and b not in memo:
-                memo.add(b)
-                yield b
-        for module in self.children():
-            for b in module._all_buffers(memo):
-                yield b
+        r"""Retained for backwards compatibility."""
```

Comment:
This can be removed if you removed all call sites from our codebase. It was an internal method.